src/ergw_aaa_diameter.erl

```diff
@@ -180,8 +182,17 @@ validate_transport(Opt, Value) ->
 validate_transport_error(Opt, Value) ->
     erlang:error(badarg, [Opt, Value]).
 
-validate_termination_cause_mapping(Opts) when is_list(Opts); is_map(Opts) ->
-    ergw_aaa_config:validate_options(fun validate_termination_cause_mapping/2, Opts, ?DEFAULT_TERMINATION_CAUSE_MAPPING);
+validate_termination_cause_mapping(Opts) when ?is_opts(Opts) ->
+    Check = case Opts of
+	    _ when is_list(Opts) -> fun(Key) -> lists:keymember(Key, 1, Opts) end;
+	    _ when is_map(Opts)  -> fun(Key) -> maps:is_key(Key, Opts) end
+    end,
+    case Check(error) of
+       false -> erlang:error(badarg, [termination_cause_mapping, Opts]);
+       true -> ok
+    end,
+    ergw_aaa_config:validate_options(
+      fun validate_termination_cause_mapping/2, Opts, ?DEFAULT_TERMINATION_CAUSE_MAPPING);
```

Comment:
not needed at all. `error` is part of `?DEFAULT_TERMINATION_CAUSE_MAPPING`, that means it can not be missing at all.
If you want to make `error` a mandatory key, the set the default for error to undefined (i.e.: `{error, undefined}`). `validate_termination_cause_mapping/2` will the report an error is the configuration does not overload error with something sensible.

Reply:
Ok. My concern is not having `error` there, so from my pov there's no need to demand one from the user as long as the default is sensible. Thanks!!!